Dockerfile

```diff
@@ -0,0 +1,52 @@
+FROM ubuntu:xenial
+
+# Default values for the build
+ARG git_branch
+ARG git_slug
+ARG c_compiler
+ARG cxx_compiler
+ARG impl
+
+RUN apt-get -yq update
+
+# Utilities
+RUN apt-get install -yq --allow-downgrades --allow-remove-essential            \
+    --allow-change-held-packages git wget apt-utils cmake unzip                \
+    libboost-all-dev software-properties-common python-software-properties libcompute-dev
+
+RUN add-apt-repository -y ppa:ubuntu-toolchain-r/test
+
+RUN apt-get -yq update
+
+# Clang 4.0
+RUN if [ "${c_compiler}" = 'clang-4.0' ]; then apt-get install -yq             \
+    --allow-downgrades --allow-remove-essential --allow-change-held-packages   \
+     clang-4.0 libomp-dev; fi
+
+# GCC 6
+RUN if [ "${c_compiler}" = 'gcc-6' ]; then apt-get install -yq                 \
+    --allow-downgrades --allow-remove-essential --allow-change-held-packages   \
+    g++-6 gcc-6; fi
+
+# OpenCL
+RUN apt-get install -yq --allow-downgrades --allow-remove-essential           \
+    --allow-change-held-packages ocl-icd-opencl-dev ocl-icd-dev opencl-headers
+
+RUN git clone https://github.com/${git_slug}.git -b ${git_branch} /SyclParallelSTL
+
```

Comment:
It would be interesting to get pocl and use it to run ComputeCpp on top, but its better if we leave it for a separate issue

Reply:
It should be possible to automatically install the Intel CPU OpenCL runtime in docker as well.

Reply:
Not sure if it will work since requires SSE extensions, are they available on travis VM?

Reply:
I can't find anything on their webpage but there seem to be github projects using SSE instructions.